refactor: fix mypy errors in cvedb.py

We're in the middle of trying to improve the type hints we use in cve-bin-tool.  I previously filed some issues as found with pyright, this particular set of issues is done using mypy.  We're new to type checking, so it's entirely possible that the types currently in the file are incorrect or incomplete. To test the file yourself, you'll need to install mypy (e.g. `pip install mypy`) and then run it with the filename you want to test.   You may need to also run `mypy --install-types` to get it to work on some files.

Note that I'm filing issues for each file individually to help folk who are looking for fun stuff to do as part of hacktoberfest so you can claim one and not interfere with anyone else working on another file.  If you've already got your 4 commits for hacktoberfest, please leave these issues for someone who's still working on theirs!

Here's a list of issues found:

```
cve_bin_tool/cvedb.py:82: error: Need type annotation for "data" (hint: "data: List[<type>] = ...")
cve_bin_tool/cvedb.py:86: error: Need type annotation for "exploits_list" (hint: "exploits_list: List[<type>] = ...")
cve_bin_tool/cvedb.py:159: error: Item "None" of "Optional[Connection]" has no attribute "cursor"
cve_bin_tool/cvedb.py:186: error: Item "None" of "Optional[Connection]" has no attribute "cursor"
cve_bin_tool/cvedb.py:241: error: Item "None" of "Optional[Connection]" has no attribute "cursor"
cve_bin_tool/cvedb.py:255: error: Item "None" of "Optional[Connection]" has no attribute "commit"
cve_bin_tool/cvedb.py:284: error: Item "None" of "Optional[Connection]" has no attribute "cursor"
cve_bin_tool/cvedb.py:294: error: Item "None" of "Optional[Connection]" has no attribute "commit"
cve_bin_tool/cvedb.py:382: error: Item "None" of "Optional[Connection]" has no attribute "cursor"
cve_bin_tool/cvedb.py:547: error: Argument 1 to "move" has incompatible type "Path"; expected "str"
```

Helpful resources:
* [Python typing documentation](https://docs.python.org/3/library/typing.html)
* [mypy cheat sheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)
* If you can't find the answer in the docs, googling the error messages above will often help!
* [cve-bin-tool's contributor docs](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md) 
* If you've contributed to open source but not this project, you might just want our [checklist for a great pull request](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#checklist-for-a-great-pull-request)

Tiny tips for new contributors:
* cve-bin-tool uses https://www.conventionalcommits.org/ style for commit messages, and we have a test that checks the title of your PR.  A good potential title for this one is in the title of this issue.
* You can make an issue auto close by including a comment "fixes #ISSUENUMBER" in your PR comments where ISSUENUMBER is the actual number of the issue.

--- a/cve_bin_tool/cvedb.py
+++ b/cve_bin_tool/cvedb.py
@@ -79,11 +79,11 @@ class CVEDB:
         self.dbpath = self.cachedir / DBNAME
         self.connection: sqlite3.Connection | None = None
 
-        self.data = []
+        self.data: list = []
         self.cve_count = -1
         self.all_cve_entries: list[dict[str, Any]] | None = None
 
-        self.exploits_list = []
+        self.exploits_list: list = []
         self.exploit_count = 0
 
         if not self.dbpath.exists():
@@ -155,8 +155,8 @@ class CVEDB:
         self.LOGGER.debug("Check database is using latest schema")
         if not cursor:
             self.db_open()
-
-        cur = self.connection.cursor()
+        if self.connection is not None:
+            cur = self.connection.cursor()
         schema_check = "SELECT * FROM cve_severity WHERE 1=0"
         result = cur.execute(schema_check)
         schema_latest = False
@@ -183,7 +183,8 @@ class CVEDB:
     def check_cve_entries(self) -> bool:
         """Report if database has some CVE entries"""
         self.db_open()
-        cursor = self.connection.cursor()
+        if self.connection is not None:
+            cursor = self.connection.cursor()
         cve_entries_check = "SELECT COUNT(*) FROM cve_severity"
         cursor.execute(cve_entries_check)
         # Find number of entries
@@ -238,7 +239,8 @@ class CVEDB:
     def init_database(self) -> None:
         """Initialize db tables used for storing cve/version data"""
         self.db_open()
-        cursor = self.connection.cursor()
+        if self.connection is not None:
+            cursor = self.connection.cursor()
 
         cve_data_create, version_range_create = self.table_schemas()
 
@@ -252,7 +254,8 @@ class CVEDB:
             self.LOGGER.info("Upgrading database to latest schema")
             cursor.execute("DROP TABLE cve_severity")
             cursor.execute(cve_data_create)
-        self.connection.commit()
+        if self.connection is not None:
+            self.connection.commit()
 
         self.db_close()
 
@@ -281,7 +284,8 @@ class CVEDB:
             severity_data, affected_data = cve_data
 
             self.db_open()
-            cursor = self.connection.cursor()
+            if self.connection is not None:
+                cursor = self.connection.cursor()
 
             if severity_data is not None:
                 self.populate_severity(severity_data, cursor, data_source=source_name)
@@ -290,8 +294,8 @@ class CVEDB:
                     affected_data,
                     cursor,
                 )
-
-            self.connection.commit()
+            if self.connection is not None:
+                self.connection.commit()
             self.db_close()
 
     def populate_severity(self, severity_data, cursor, data_source):
@@ -379,7 +383,8 @@ class CVEDB:
         Fetches vendor from the database for packages that doesn't have vendor info for Package List Parser Utility and Universal Python package checker.
         """
         self.db_open()
-        cursor = self.connection.cursor()
+        if self.connection is not None:
+            cursor = self.connection.cursor()
         vendor_package_pairs = []
         query = """
         SELECT DISTINCT vendor FROM cve_range

--- a/cve_bin_tool/cvedb.py
+++ b/cve_bin_tool/cvedb.py
@@ -79,11 +79,11 @@ class CVEDB:
         self.dbpath = self.cachedir / DBNAME
         self.connection: sqlite3.Connection | None = None
 
-        self.data: list = []
+        self.data: list[Any] = []
         self.cve_count = -1
         self.all_cve_entries: list[dict[str, Any]] | None = None
 
-        self.exploits_list: list = []
+        self.exploits_list: list[Any] = []
         self.exploit_count = 0
 
         if not self.dbpath.exists():

